crates/apollo-compiler/src/name.rs

```diff
@@ -0,0 +1,511 @@
+    }
+}
+
+impl Ord for Name {
+    #[inline]
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.as_str().cmp(other.as_str())
+    }
+}
+
+impl PartialOrd for Name {
+    #[inline]
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl PartialEq<str> for Name {
+    #[inline]
+    fn eq(&self, other: &str) -> bool {
+        self.as_str() == other
+    }
+}
+
+impl PartialOrd<str> for Name {
+    #[inline]
+    fn partial_cmp(&self, other: &str) -> Option<std::cmp::Ordering> {
+        self.as_str().partial_cmp(other)
+    }
+}
+
+impl PartialEq<&'_ str> for Name {
+    #[inline]
+    fn eq(&self, other: &&'_ str) -> bool {
+        self.as_str() == *other
+    }
+}
+
+impl PartialOrd<&'_ str> for Name {
+    #[inline]
+    fn partial_cmp(&self, other: &&'_ str) -> Option<std::cmp::Ordering> {
+        self.as_str().partial_cmp(*other)
+    }
+}
+
+impl From<&'_ Self> for Name {
+    #[inline]
+    fn from(value: &'_ Self) -> Self {
+        value.clone()
+    }
+}
+
+impl serde::Serialize for Name {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.as_str())
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for Name {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        const EXPECTING: &str = "a string in GraphQL Name syntax";
+        struct Visitor;
+        impl<'de> serde::de::Visitor<'de> for Visitor {
+            type Value = Name;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str(EXPECTING)
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Name::new(v)
+                    .map_err(|_| E::invalid_value(serde::de::Unexpected::Str(v), &EXPECTING))
+            }
+        }
+        deserializer.deserialize_str(Visitor)
+    }
+}
+
+impl TryFrom<&str> for Name {
+    type Error = InvalidNameError;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        Self::new(value)
+    }
+}
+
+impl TryFrom<String> for Name {
+    type Error = InvalidNameError;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        Self::new(&value)
+    }
+}
+
+impl TryFrom<&'_ String> for Name {
+    type Error = InvalidNameError;
+
+    fn try_from(value: &'_ String) -> Result<Self, Self::Error> {
+        Self::new(value)
+    }
+}
+
+impl AsRef<Name> for Name {
+    fn as_ref(&self) -> &Name {
+        self
+    }
+}
+
+impl ToCliReport for InvalidNameError {
+    fn location(&self) -> Option<NodeLocation> {
+        self.location
+    }
+    fn report(&self, report: &mut CliReport) {
+        report.with_label_opt(self.location, "cannot be parsed as a GraphQL name");
```

Comment:
capitalising to indicate somehow it's a token
```suggestion
        report.with_label_opt(self.location, "cannot be parsed as a GraphQL Name");
```